concourse/scripts/environment.sh

```diff
@@ -17,15 +17,40 @@ fi
 
 AWS_ACCOUNT=${AWS_ACCOUNT:-dev}
 
-CONCOURSE_URL="${CONCOURSE_URL:-http://localhost:8080}"
-FLY_TARGET="${FLY_TARGET:-${DEPLOY_ENV}-bootstrap}"
-FLY_CMD="${PROJECT_DIR}/bin/fly-bootstrap"
-
+BOSH_FQDN="bosh.${SYSTEM_DNS_ZONE_NAME}"
+BOSH_FQDN_EXTERNAL="bosh-external.${SYSTEM_DNS_ZONE_NAME}"
 CONCOURSE_ATC_USER=${CONCOURSE_ATC_USER:-admin}
-if [ -z "${CONCOURSE_ATC_PASSWORD:-}" ]; then
-    user_id=$(aws sts get-caller-identity | awk '$1 ~ /UserId/ {print $2}')
-    CONCOURSE_ATC_PASSWORD=$(hashed_password "${user_id}")
-fi
+case "${TARGET_CONCOURSE}" in
+  bootstrap)
+    CONCOURSE_URL="${CONCOURSE_URL:-http://localhost:8080}"
+    FLY_TARGET="${FLY_TARGET:-${DEPLOY_ENV}-bootstrap}"
+    FLY_CMD="${PROJECT_DIR}/bin/fly-bootstrap"
+
+    if [ -z "${CONCOURSE_ATC_PASSWORD:-}" ]; then
+        user_id=$(aws sts get-caller-identity | awk '$1 ~ /UserId/ {print $2}')
+        CONCOURSE_ATC_PASSWORD=$(hashed_password "${user_id}")
+    fi
+    BOSH_LOGIN_HOST=${BOSH_FQDN_EXTERNAL}
+    ;;
+  deployer-concourse|build-concourse)
+    if [ "$TARGET_CONCOURSE" == 'deployer-concourse' ]; then
+      CONCOURSE_URL="${CONCOURSE_URL:-"https://deployer.${SYSTEM_DNS_ZONE_NAME}"}"
+    else
+      CONCOURSE_URL="${CONCOURSE_URL:-"https://concourse.${SYSTEM_DNS_ZONE_NAME}"}"
```

Comment:
This logic is already captured by the `CONCOURSE_HOSTNAME` variable set in the `Makefile`. Would it be possible to re-use that instead of having to repeat the information here.